### PR TITLE
Improve TRC20 token transfer message

### LINKED ISSUE #643: Add TRC20 logic to wallet-core and review other message to reduce integration complexity.
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by leoneparise**
## Description
Fixes #643 
Improve TRC20 token transfer message and set defaults for timestamp and expiration when not set.

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: src/Tron/Signer.cpp [hunk 1]
```diff
@@ -257,10 +279,29 @@ Proto::SigningOutput Signer::sign(const Proto::SigningInput& input) noexcept {
...
 
-    internal.mutable_raw_data()->set_timestamp(input.transaction().timestamp());
-    internal.mutable_raw_data()->set_expiration(input.transaction().expiration());
+    // Get default timestamp and expiration
+    const uint64_t now = duration_cast< milliseconds >(
+            system_clock::now().time_since_epoch()
+    ).count();
+    const uint64_t timestamp = input.transaction().timestamp() == 0
+            ? now
+            : input.transaction().timestamp();
```
Comment (vikmeup) [inline src/Tron/Signer.cpp]: it think this logic might be an extra, I would just use what's provided by the client

Comment (hewigovens) [reply src/Tron/Signer.cpp]: Yes, just use timestamp from app

File: src/Tron/Signer.cpp [hunk 2]
```diff
@@ -150,6 +152,26 @@ protocol::TriggerSmartContract to_internal(const Proto::TriggerSmartContract& tr
     return internal;
 }
 
+protocol::TriggerSmartContract to_internal(const Proto::TransferTRC20Contract& transferTrc20Contract) {
+    auto toAddress = Base58::bitcoin.decodeCheck(transferTrc20Contract.to_address());
+    Data amount;
+    encode64BE(transferTrc20Contract.amount(), amount);
+
+    // Encode smart contract call parameters
+    auto contract_params = parse_hex("0xa9059cbb");
```
Comment (hewigovens) [inline src/Tron/Signer.cpp]: Can we use constants for `0xa9059cbb`?

File: src/Data.h [hunk 3]
```diff
@@ -15,6 +15,10 @@ namespace TW {
 using byte = std::uint8_t;
 using Data = std::vector<byte>;
 
+inline void pad_left(Data& data, const uint32_t size) {
+    data.insert(data.begin(), size - data.size(), 0);
```
Comment (hewigovens) [inline src/Data.h]: do we need to check `size - data.size()`?

Comment (leoneparise) [reply src/Data.h]: Yes, we have to add `size - data.size()` zeros to the vector. If you have a better way to do it, let me know.

File: src/Tron/Signer.cpp (hunk 1 above)
Comment (leoneparise) [reply src/Tron/Signer.cpp]: The timestamp is only set if the app doesn't provide it. Same for expiration,

File: src/Tron/Signer.cpp (hunk 2 above)
Comment (leoneparise) [reply src/Tron/Signer.cpp]: Sure

**Pull request merged**